### PR TITLE
2808 - introduced es6 to the component

### PR DESCRIPTION
**Pull request by eleanor-heath**
https://github.com/adaptlearning/adapt_framework/issues/2808

### CODE REVIEW

**Review by cahirodoherty-learningpool — APPROVED** [review]
👀

**Review by ethan-lp — APPROVED** [review]

File: js/adapt-contrib-componentName.js [hunk 1]
```diff
@@ -1,33 +1,32 @@
...
+    this.setReadyStatus();
+
+    // Use this to set the model status to complete.
+    // This can be used with inview or when the model is set to complete/the question has been answered.
+    this.setCompletionStatus();
+  }
+
+}
+
+export default Adapt.register('componentName', ComponentName);
```
Comment (oliverfoster) [inline js/adapt-contrib-componentName.js]: Move to a separate file as with other plugins. The view file should be called `ComponentNameView.js` and `export default ComponentNameView;` here instead.

File: js/adapt-contrib-componentName.js [hunk 2]
```diff
@@ -1,33 +1,32 @@
...
-});
\ No newline at end of file
+import Adapt from 'core/js/adapt';
+import ComponentView from 'core/js/views/componentView';
+
+class ComponentName extends ComponentView {
+
+  get template() {
+    return 'componentName';
+  }
```
Comment (oliverfoster) [inline js/adapt-contrib-componentName.js]: At the bottom `ComponentNameView.template = 'componentName';`

File: js/adapt-contrib-componentName.js [hunk 3]
```diff
@@ -1,33 +1,32 @@
...
-        template: "componentName"
-    });
-
-    return Adapt.register("componentName", ComponentName);
-});
\ No newline at end of file
+import Adapt from 'core/js/adapt';
+import ComponentView from 'core/js/views/componentView';
+
+class ComponentName extends ComponentView {
```
Comment (oliverfoster) [inline js/adapt-contrib-componentName.js]:
```suggestion
class ComponentNameView extends ComponentView {
```

**Review by oliverfoster — CHANGES_REQUESTED** [review]

File: js/adapt-contrib-componentName.js [hunk 4]
```diff
@@ -1,33 +1,8 @@
...
-
-    }, {
-        template: "componentName"
-    });
-
-    return Adapt.register("componentName", ComponentName);
-});
\ No newline at end of file
+import Adapt from 'core/js/adapt';
+import ComponentNameModel from './ComponentNameModel';
```
Comment (oliverfoster) [inline js/adapt-contrib-componentName.js:2]: You'll need to add the file `js/ComponentNameModel.js`

Something like:
```js
import ComponentModel from 'core/js/models/componentModel';

export class ComponentNameModel extends ComponentModel {}
```

see graphic

**Review by oliverfoster — CHANGES_REQUESTED** [review]

**Review by oliverfoster — APPROVED** [review]

**Pull request merged**